rainier-core/src/main/scala/com/stripe/rainier/core/RandomVariable.scala

```diff
@@ -42,9 +42,20 @@ class RandomVariable[+T](val value: T,
     sampleable.get(value)(rng, num)
 
   def sample[V]()(implicit rng: RNG, sampleable: Sampleable[T, V]): List[V] =
-    sample(Sampler.Default.sampler,
-           Sampler.Default.warmupIterations,
-           Sampler.Default.iterations)
+    sample(Sampler.Default.iterations)
+
+  def sample[V](iterations: Int)(implicit rng: RNG,
+                                 sampleable: Sampleable[T, V]): List[V] =
+    if (density.variables.isEmpty) {
+      val fn = sampleable.prepare(value, density.variables)
```

Comment:
hmm, would it make sense to avoid the warmup within the more generic version of `sample` below? then this one can just call `sample(...)` with all the defaults.